polling_stations/apps/data_importers/data_types.py

```diff
@@ -0,0 +1,134 @@
+"""
+Data type classes used by base importers
+"""
+
+import abc
+import logging
+
+from addressbase.models import get_uprn_hash_table, UprnToCouncil
+from uk_geo_utils.helpers import Postcode
+
+
+class CustomSet(metaclass=abc.ABCMeta):
+    def __init__(self):
+        self.elements = set()
+
+    def add(self, element):
+        self.elements.add(self.build_namedtuple(element))
+
+    @abc.abstractmethod
+    def build_namedtuple(self, element):
+        pass
+
+
+class AssignPollingStationsMixin(metaclass=abc.ABCMeta):
+    @abc.abstractmethod
+    def get_polling_station_lookup(self):
+        pass
+
+    @property
+    def council_id(self):  # TODO Deal with old_to_new council_ids map
```

Comment:
Unfortunately, I think the reverse is true.
At least with GSS codes they have a predictable publication schedule and when changes are upcoming we can get hold of the new GSS codes ahead of time. It would be useful if they were actually _published_ ahead of time, but at least ONS Geography support are really helpful (and consistently have been for many years). Using register codes makes changes much harder to deal with change because you won't be able to get hold of the codes for new areas ahead of time, there is no transparency or regularity when it comes to when (or if) they are published and there's nobody helpful we can contact about them.

Reply:
I guess that now register codes are coming via the EC, if they don't have one, then we need to get them to put in place a temporary one. I guess this isn't ideal, but it doesn't stop us updating GSS codes (read: making sure EC update GSS codes) and still using them for imports.

Reply:
From experience, you will probably be leading the Commission on this rather than the other way around.